--- a/src/lib/epddl/grounder/initial_state/explicit_initial_state_grounder.cpp	2026-02-26 16:06:39.826356744 -0600
+++ b/src/lib/epddl/grounder/initial_state/explicit_initial_state_grounder.cpp	2026-08-13 22:24:53.957178352 -0600
@@ -103,7 +103,7 @@
             l->get_predicates(), ground_elem, info, info.context.types.get_type("object"));
 
     for (const del::atom p : l_atoms)
-        ground_atoms.push_back(p);
+        ground_atoms.set(p);
 
     return del::label{std::move(ground_atoms)};
 }
